pxd_bio_blkmq.c

```diff
@@ -577,25 +600,45 @@ static void fp_handle_specialops(struct work_struct *work) {
         BUG_ON(!rq_is_special(rq));
         atomic_inc(&pxd_dev->fp.nio_discard);
 
-        // submit discard to replica
-        if (blk_queue_discard(q)) { // discard supported
-                r = blkdev_issue_discard(bdev, blk_rq_pos(rq),
-                                         blk_rq_sectors(rq), GFP_NOIO, 0);
-        } else if (bdev_write_same(bdev)) {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,19,0)
```

Comment:
below 2 sections do not consider write_same availability, instead just straight to zero out.
Looks ok, as most of the devices will support discard.

Reply:
Ok, let me take a look at it separately if we need to do something here.
Will merge it for now.